client/signup/steps/site-style/index.jsx

```diff
@@ -0,0 +1,181 @@
+		this.state = {
+			siteStyle: selectedStyle.id,
+			themeSlugWithRepo: selectedStyle.theme,
+		};
+	}
+
+	componentDidMount() {
+		SignupActions.saveSignupStep( {
+			stepName: this.props.stepName,
+		} );
+	}
+
+	handleStyleOptionChange = event => {
+		const selectedStyle = find( this.props.styleOptions, [ 'id', event.currentTarget.value ] );
+		this.setState(
+			{
+				siteStyle: selectedStyle.id,
+				themeSlugWithRepo: selectedStyle.theme,
+			},
+			() => this.props.setSiteStyle( selectedStyle.id )
+		);
+	};
+
+	handleSubmit = event => {
+		event.preventDefault();
+		this.props.submitSiteStyle( this.state );
+	};
+
+	renderStyleOptions() {
+		return this.props.styleOptions.map( siteStyleProperties => {
+			const isChecked = siteStyleProperties.id === this.state.siteStyle;
+			const optionLabelClasses = classNames( 'site-style__option-label', {
+				'is-checked': isChecked,
+				[ `site-style__variation-${ siteStyleProperties.id }` ]: siteStyleProperties.id,
+			} );
+			return (
+				<FormLabel
+					htmlFor={ siteStyleProperties.id }
+					className={ optionLabelClasses }
+					key={ siteStyleProperties.id }
+					title={ siteStyleProperties.description || '' }
+				>
+					<FormRadio
+						id={ siteStyleProperties.id }
+						value={ siteStyleProperties.id }
+						name="site-style-option"
+						checked={ isChecked }
+						onChange={ this.handleStyleOptionChange }
+					/>
+					<span>
+						<strong>{ siteStyleProperties.label }</strong>
+					</span>
+				</FormLabel>
+			);
+		} );
+	}
+
+	renderContent = () => (
+		<div className="site-style__wrapper">
+			<Card>
+				<div className="site-style__form-wrapper">
+					<form className="site-style__form" onSubmit={ this.handleSubmit }>
+						<FormFieldset className="site-style__fieldset">
+							{ this.renderStyleOptions() }
+						</FormFieldset>
+						<div className="site-style__submit-wrapper">
+							<Button primary={ true } type="submit">
+								{ this.props.translate( 'Continue' ) }
+							</Button>
+						</div>
+					</form>
+				</div>
+			</Card>
+		</div>
+	);
+
+	render() {
+		const { flowName, positionInFlow, signupProgress, stepName, translate } = this.props;
+		const headerText = translate( 'Choose a style' );
+		const subHeaderText = translate(
+			"Choose a style for your site's theme. Don't worry, you can always change it later."
+		);
+
+		return (
+			<div>
+				<StepWrapper
+					flowName={ flowName }
+					stepName={ stepName }
+					positionInFlow={ positionInFlow }
+					headerText={ headerText }
+					fallbackHeaderText={ headerText }
+					subHeaderText={ subHeaderText }
+					fallbackSubHeaderText={ subHeaderText }
+					signupProgress={ signupProgress }
+					stepContent={ this.renderContent() }
+				/>
+			</div>
+		);
+	}
+}
+
+const mapDispatchToProps = ( dispatch, ownProps ) => ( {
+	submitSiteStyle: ( { siteStyle, themeSlugWithRepo } ) => {
+		const { flowName, stepName, goToNextStep } = ownProps;
```

Comment:
After that, we simply derive `themeSlugWithRepo` with `props.styleOptions` and `props.siteStyle` here.